Feature/psr4

chilimatic opened the pull request:
this should be psr4 compatible now ..... pls review and test it.

chilimatic committed:
- psr4 like structure

chilimatic committed:
- fixed templates
- fixed initial state asumption
- reduced the autoloader function

chilimatic committed:
- moved autoloader to composer psr4
- fixed namespace issues

andreas83 commented:
Wohoo looks really good.
I will test and merge it as soon as possible.

andreas83 commented:
i get the following error:
Fatal error: Class '\SocialNetwork\app\controller\DataController' not found in SocialNetwork/routing.php on line 105

chilimatic commented:
you triggered a composer update ? so the autoloader is registered ?

andreas83 commented:
sorry my fault !

andreas83 commented:
okay looks like most parts works again, except:
 Call to undefined function SocialNetwork\app\lib\transformer\string\html\YoutubeTransformer() in <b>/SocialNetwork/app/lib/transformer/TransformerFactory.php</b> on line <b>63</b><br />

andreas83 commented:
Found another one:
Fatal error: Call to a member function fetchALL() on boolean in /home/lissi/dev/SocialNetwork/app/lib/BaseModel.php on line 5

Reason : namespace is now included in $this->table 

SELECT \* FROM SocialNetwork\app\model\Content WHERE id = 356

chilimatic committed:
- new abstract getSource Method which returns the tablename
- new className variable so the classname is only fetched once
- fixed property usages

chilimatic commented:
seperated the object name from the table name should work now

andreas83 commented:
Yepp the getSource method works!
Any idea whats wrong with the TransformerFactory ?

andreas83 commented:
mhm another question, is the app/lib/AutoLoader.php missing ?

chilimatic commented:
yes since it's psr4 the composer autoloader can load the classes. But you can take the file from 

https://github.com/andreas83/SocialNetwork/pull/60/commits/b3a4c33ace2deeb4ac837521f4f6eb1770662147

and build your own psr4 autoloader out of it  [it just misses the prefix and the basedir map which can be easily added]. 

But since this project is heavily dependent of composer I don't really see the benefit.

andreas83 commented:
The reason why i asked was, because i thought its required since you added the 
require_once 'app/lib/AutoLoader.php'; to some files....

I also see no reason for a separate auto loader since it can be done via autoloader from composer

Found some more issues
- [x] Fatal error: Class 'SocialNetwork\app\lib\User' not found in /home/lissi/dev/SocialNetwork/app/lib/Helper.php on line 82
- [x] Call to undefined function SocialNetwork\app\lib\transformer\string\html\YoutubeTransformer() in /SocialNetwork/app/lib/transformer/TransformerFactory.php on line 63

chilimatic committed:
- fixed static Factory pattern

chilimatic commented:
lol :D andi !!! <3 I forgot the "new" keyword :D sorry ....
Call to undefined function SocialNetwork\app\lib\transformer\string\html\YoutubeTransformer()

andreas83 commented:
you are my hero ! :kiss: 

shame on me, i should be able to fix such bugs in minutes

andreas83 commented:
makeStatic and make looks 99% identical (possible code smell ?)

chilimatic committed:
DRY

chilimatic commented:
ah yes thx.

andreas83 commented on app/lib/transformer/TransformerFactory.php:
```diff
@@ -50,19 +50,7 @@ public static function makeStatic($name)
-        }
-        $className = __NAMESPACE__ . '\\' . $name;
-
-        self::$runtimeCache[$name] = new $className();
```
Now makeStatic does not create a _new_ instance which was the 1 percent difference

andreas83 commented:
http://my url /api/metadata/?url=https://www.youtube.com/watch?v=PUiaKjVpP8M

For some very unknown reasons VideoTransformerTrait::transform never gets called

chilimatic committed:
- forgot the new instance

chilimatic committed:
- fixed return value for factory

chilimatic commented:
sry, fixed :)

andreas83 committed:
removed old autoloading and replaced with new psr-4 composer autoloading

andreas83 committed:
fixed automated generation of insert statements

andreas83 commented:
wohooo  looks good, i will merge everything to upstream
